webpack.common.js

```diff
@@ -0,0 +1,57 @@
+const path = require("path");
+const { CleanWebpackPlugin } = require("clean-webpack-plugin");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+
+const public = path.resolve(__dirname, "public");
+
+module.exports = {
+  entry: "./src/index.tsx",
+  output: {
+    filename: "[name].[contenthash].js",
+    path: path.resolve(__dirname, "build"),
```

Comment:
Resolve this path in var on top, and use it here as you've done with the public path.
```
const __builddir = path.resolve(__dirname, "build");
```